### PR TITLE
Updated localnet to use more recent commits of optimism and op-geth

### PR DESCRIPTION
**Pull request by ClaytonNorthey92**
**Summary**
Updated localnet to use more recent commits of optimism and op-geth

**Changes**
updated localnet op-stack dependencies to use more recent commits

using make dev-allocs for l1 deployments

deleted deployments directory

### CODE REVIEW

**Comment by ClaytonNorthey92** [comment]
depends on https://github.com/hemilabs/op-geth/pull/6

**Comment by ClaytonNorthey92** [comment]
fixes https://github.com/hemilabs/heminetwork/issues/127

File: e2e/docker-compose.yml [hunk 1]
```diff
@@ -192,6 +194,7 @@ services:
       - "./deploy-config.json:/git/optimism/packages/contracts-bedrock/deploy-config/devnetL1.json"
       - "./prestate-proof.json:/git/optimism/op-program/bin/prestate-proof.json"
       - { type: "tmpfs", target: "/tmp" }
+    network_mode: host
```
Comment (joshuasing) [inline e2e/docker-compose.yml:197]: What is the advantage for using host networking for this? 🤔

File: e2e/entrypointl2.sh [hunk 2]
```diff
@@ -36,7 +36,10 @@ sh /tmp/genesisl2.sh
  /tmp/datadir \
  --authrpc.vhosts="*" \
  --authrpc.addr=0.0.0.0 \
- --authrpc.port=8551 \
+--authrpc.port=8551 \
```
Comment (joshuasing) [inline e2e/entrypointl2.sh:39]: I think the indentation of this line was changed 🤔

File: e2e/optimism-stack.Dockerfile [hunk 3]
```diff
@@ -20,38 +20,42 @@ ENV PATH="${PATH}:/root/.foundry/bin"
 
 RUN foundryup
 
-ARG OP_GETH_CACHE_BREAK=1
+ARG OP_GETH_CACHE_BREAK=4
```
Comment (joshuasing) [inline e2e/optimism-stack.Dockerfile:23]: Was this intentionally changed?

File: e2e/optimism-stack.Dockerfile [hunk 4]
```diff
@@ -20,38 +20,42 @@ ENV PATH="${PATH}:/root/.foundry/bin"
...
 RUN git clone https://github.com/hemilabs/op-geth
 WORKDIR /git/op-geth
-RUN git checkout a012302a04f050d09c11a8fd5deb630ab7a376ad
+RUN git checkout 83dd8a6e0459dd0f96182bb16065411f9318ac00
 
 WORKDIR /git
 
-ARG OPTIMISM_CACHE_BREAK=1
+ARG OPTIMISM_CACHE_BREAK=2
```
Comment (joshuasing) [inline e2e/optimism-stack.Dockerfile:30]: Was this intentionally changed?

File: e2e/optimism-stack.Dockerfile (hunk 3 above)
Comment (ClaytonNorthey92) [reply e2e/optimism-stack.Dockerfile:23]: haha I thought no one would notice, sort of;  the value doesn't mean much (just an easy way to break the cache which you can override in the cli, I was being lazy)

File: e2e/entrypointl2.sh (hunk 2 above)
Comment (ClaytonNorthey92) [reply e2e/entrypointl2.sh:39]: ah nope will fix

**Review by joshuasing — APPROVED** [review]

**Review by marcopeereboom — APPROVED** [review]

**Review by jcvernaleo — APPROVED** [review]
OK

**Pull request merged**